modules/ocf_mirrors/files/collect-mirrors-stats

```diff
@@ -0,0 +1,168 @@
+            dist = stats[6]
+            dist = dist.split('/')[1] if '/' in dist else dist
+
+            # record if we returned http 2xx/3xx
+            if stats[8][0] in ('2', '3') and dist in projects:
+                projects[dist]['up'] += int(stats[-2])
+                projects[dist]['down'] += int(stats[-1])
+            else:
+                projects['other']['up'] += int(stats[-2])
+                projects['other']['down'] += int(stats[-1])
+
+    return projects
+
+
+def process_rsync_log(filename):
+    """Parse Tx/Rx information out of an rsyncd log file.
+
+    This function is project-agnostic.
+    """
+
+    tx = 0
+    rx = 0
+    with filename.open('r') as f:
+        for line in f.readlines():
+            # log format is <datetime> <pid> sent <tx> bytes received <rx> bytes total size
+            if line.find('sent') > 0:
+                stats = line.split()
+                tx += int(stats[4])
+                rx += int(stats[7])
+
+    return tx, rx
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Process mirrors logs to calculate network usage '
+                                                 'and store in ocfstats')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='do not print stats after collecting them')
+    parser.add_argument('-n', '--dry-run', action='store_true',
+                        help='dry-run mode, do not write statistics to the database')
+    parser.add_argument('--apache', nargs='*',
+                        help='Apache log file(s) to process')
+    parser.add_argument('--rsync', nargs='*',
+                        help='rsync log files(s) to process')
+    parser.add_argument('date', nargs='?', default=date.today() - timedelta(1),
+                        help='date for use in filtering log entries')
+
+    args = parser.parse_args()
+
+    # first, figure out what projects we're getting stats for
+    sources = []
+    for project in MIRRORS_DATA_PATH.iterdir():
+        if project.is_dir() and not project.name.startswith('.'):
+            sources.append(project.name)
+
+    sources.append('other')  # catchall
+
+    # then, figure out which Apache logs we're pulling stats from
+    if args.apache:
+        apache_log_files = args.apache
+    else:
+        apache_log_files = []
+        for log_name in APACHE_LOG_FILES:
+            # logrotate rotates the log at 6am, but this script
+            # runs at midnight. So to capture 24h of data, we need
+            # to parse the rotated log as well. (the .1 file)
+            log = APACHE_LOG_PATH / log_name
+            apache_log_files.append(log)
+            apache_log_files.append(Path(str(log) + '.1'))
+
+    # and the same for rsync
+    if args.rsync:
+        rsync_log_files = args.rsync
+    else:
+        rsync_log_files = []
+        for log_file in RSYNC_LOG_PATH.iterdir():
+            # log files are named <project>.log
```

Comment:
Why do we have this comment? Did you mean to add an `if log_file.ends_with('.log')`?

Reply:
no, it's just a comment so the reader knows what the contents of the list look like